controllers/argocd/deployment.go

```diff
@@ -377,7 +377,10 @@ func (r *ReconcileArgoCD) reconcileDexDeployment(cr *argoprojv1a1.ArgoCD) error
 	if dexDisabled {
 		log.Info("reconciling for dex, but dex is disabled")
 	}
-
+	if cr.Spec.NodePlacement != nil {
```

Comment:
Since we are adding the `NodePlacement` spec for all deployment types, is it OK if we add it within the `newDeploymentWithName` func?
Haven't evaluated this completely though, but it would make sense IMO. That way we don't have to explicitly add this for each deployment type.

Reply:
Thanks a lot for this, makes the code a lot more leaner :)